--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,10 +21,11 @@
 include(ECMInstallIcons)
 include(FeatureSummary)
 include(ECMSetupVersion)
 include(CMakeFindDependencyMacro)
 include(ECMQtDeclareLoggingCategory)
+include(FindPkgConfig)
 
 find_package(Qt${QT_MAJOR_VERSION} ${QT_MIN_VERSION} CONFIG REQUIRED COMPONENTS
   Core
   DBus
   Gui
@@ -56,11 +57,11 @@
   AkonadiContact
   Contacts
 )
 
 # only headers are used
-find_package(Gpgme REQUIRED)
+pkg_search_module(GPGME REQUIRED gpgme)
 
 add_definitions(
     #-DQT_DISABLE_DEPRECATED_BEFORE=0x050f00
     -DQT_DEPRECATED_WARNINGS_SINCE=0x060000
     -DKF_DISABLE_DEPRECATED_BEFORE_AND_AT=0x055200
