From: Nicolas Braud-Santoni <nicolas@braud-santoni.eu>
Date: Thu, 28 Jul 2016 16:49:18 +0200
Subject: Set log_group=adm
Forwarded: not-needed

---
 init.d/auditd.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/init.d/auditd.conf
+++ b/init.d/auditd.conf
@@ -5,7 +5,7 @@
 local_events = yes
 write_logs = yes
 log_file = /var/log/audit/audit.log
-log_group = root
+log_group = adm
 log_format = ENRICHED
 flush = INCREMENTAL_ASYNC
 freq = 50
