Upstream uses prefix= while we use prefix=/usr

Index: glibc-2.27/sysdeps/mach/hurd/i386/libc.abilist
===================================================================
--- glibc-2.27.orig/sysdeps/mach/hurd/i386/libc.abilist
+++ glibc-2.27/sysdeps/mach/hurd/i386/libc.abilist
@@ -641,7 +641,7 @@ GLIBC_2.2.6 _libc_intl_domainname D 0x5
 GLIBC_2.2.6 _longjmp F
 GLIBC_2.2.6 _mcleanup F
 GLIBC_2.2.6 _mcount F
-GLIBC_2.2.6 _nl_default_dirname D 0xe
+GLIBC_2.2.6 _nl_default_dirname D 0x12
 GLIBC_2.2.6 _nl_domain_bindings D 0x4
 GLIBC_2.2.6 _nl_msg_cat_cntr D 0x4
 GLIBC_2.2.6 _null_auth D 0xc
Index: glibc-2.38/sysdeps/mach/hurd/i386/libc.abilist
===================================================================
--- glibc-2.38/sysdeps/mach/hurd/x86_64/libc.abilist.original2	2024-02-19 00:38:23.000000000 +0000
+++ glibc-2.38/sysdeps/mach/hurd/x86_64/libc.abilist	2024-02-19 00:38:30.000000000 +0000
@@ -608,7 +608,7 @@
 GLIBC_2.38 _longjmp F
 GLIBC_2.38 _mcleanup F
 GLIBC_2.38 _mcount F
-GLIBC_2.38 _nl_default_dirname D 0xe
+GLIBC_2.38 _nl_default_dirname D 0x12
 GLIBC_2.38 _nl_domain_bindings D 0x8
 GLIBC_2.38 _nl_msg_cat_cntr D 0x4
 GLIBC_2.38 _obstack_allocated_p F
