From a3ae33bc76fff1db3aa76a153e77154cb7f3bd46 Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Wed, 6 Dec 2023 00:34:52 +0100
Subject: [PATCH] common.postinst: only build AUTOINSTALL="yes" modules on
 installation

---
 dkms_common.postinst.in | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/dkms_common.postinst.in b/dkms_common.postinst.in
index e6c430d..e4065ac 100644
--- a/dkms_common.postinst.in
+++ b/dkms_common.postinst.in
@@ -155,6 +155,13 @@ if [ -z "$autoinstall" ]; then
     exit 0
 fi
 
+dkms_conf="/var/lib/dkms/$NAME/$VERSION/source/dkms.conf"
+autoinstall=$(bash -c 'AUTOINSTALL=; . "'"$dkms_conf"'" >/dev/null 2>&1; echo $AUTOINSTALL')
+if [ -z "$autoinstall" ]; then
+    echo "Not building the $NAME module which does not have AUTOINSTALL enabled."
+    exit 0
+fi
+
 # On 1st installation, let us look for a directory
 # in @MODDIR@ which matches $(uname -r). If none
 # is found it is possible that buildd is being used
-- 
2.39.5

