diff -Naur a/src/db/vnotedbmanager.cpp b/src/db/vnotedbmanager.cpp
--- a/src/db/vnotedbmanager.cpp	2022-01-18 11:15:45.450877579 +0800
+++ b/src/db/vnotedbmanager.cpp	2022-01-18 11:16:34.019109082 +0800
@@ -242,7 +242,7 @@
     dbDir.setFile(vnoteDatabasePath+QDir::separator());
 
     if (!dbDir.exists()) {
-        QDir(dbDir.filePath()).mkdir(dbDir.filePath());
+        QDir(dbDir.filePath()).mkpath(dbDir.filePath());
         qInfo() << "Create vnote db directory:" << vnoteDatabasePath;
     }
