Description: patch docs to correct path to ssh-askpass
Author: Marc Haber <mh+debian-packages@zugschlus.de>
Forwarded: not-needed
--- a/docs/UPGRADE.md
+++ b/docs/UPGRADE.md
@@ -426,11 +426,11 @@ Notes on upgrading from an older release
 
    Old style in `/etc/sudoers`:
 
-        Defaults askpass=/usr/X11R6/bin/ssh-askpass
+        Defaults askpass=/usr/bin/ssh-askpass
 
    New style in `/etc/sudo.conf`:
 
-        Path askpass /usr/X11R6/bin/ssh-askpass
+        Path askpass /usr/bin/ssh-askpass
 
  * Upgrading from a version prior to 1.7.5:
 
--- a/docs/sudo.conf.man.in
+++ b/docs/sudo.conf.man.in
@@ -238,7 +238,7 @@ For example:
 .RS 4n
 Path intercept @intercept_file@
 Path noexec @noexec_file@
-Path askpass /usr/X11R6/bin/ssh-askpass
+Path askpass /usr/bin/ssh-askpass
 .RE
 .fi
 .PP
@@ -784,7 +784,7 @@ front-end configuration
 # own askpass program but can use the OpenSSH askpass.
 #
 # Use the OpenSSH askpass
-#Path askpass /usr/X11R6/bin/ssh-askpass
+#Path askpass /usr/bin/ssh-askpass
 #
 # Use the Gnome OpenSSH askpass
 #Path askpass /usr/libexec/openssh/gnome-ssh-askpass
--- a/docs/sudo.conf.mdoc.in
+++ b/docs/sudo.conf.mdoc.in
@@ -218,7 +218,7 @@ For example:
 .Bd -literal -offset 4n
 Path intercept @intercept_file@
 Path noexec @noexec_file@
-Path askpass /usr/X11R6/bin/ssh-askpass
+Path askpass /usr/bin/ssh-askpass
 .Ed
 .Pp
 If no path name is specified, features relying on the specified
@@ -716,7 +716,7 @@ front-end configuration
 # own askpass program but can use the OpenSSH askpass.
 #
 # Use the OpenSSH askpass
-#Path askpass /usr/X11R6/bin/ssh-askpass
+#Path askpass /usr/bin/ssh-askpass
 #
 # Use the Gnome OpenSSH askpass
 #Path askpass /usr/libexec/openssh/gnome-ssh-askpass
--- a/docs/sudo.man.in
+++ b/docs/sudo.man.in
@@ -205,7 +205,7 @@ For example:
 .sp
 .RS 12n
 # Path to askpass helper program
-Path askpass /usr/X11R6/bin/ssh-askpass
+Path askpass /usr/bin/ssh-askpass
 .RE
 .fi
 .RS 8n
--- a/docs/sudo.mdoc.in
+++ b/docs/sudo.mdoc.in
@@ -207,7 +207,7 @@ used.
 For example:
 .Bd -literal -offset 4n
 # Path to askpass helper program
-Path askpass /usr/X11R6/bin/ssh-askpass
+Path askpass /usr/bin/ssh-askpass
 .Ed
 .Pp
 If no askpass program is available,
--- a/examples/sudo.conf.in
+++ b/examples/sudo.conf.in
@@ -24,7 +24,7 @@
 # own askpass program but can use the OpenSSH askpass.
 #
 # Use the OpenSSH askpass
-#Path askpass /usr/X11R6/bin/ssh-askpass
+#Path askpass /usr/bin/ssh-askpass
 #
 # Use the Gnome OpenSSH askpass
 #Path askpass /usr/libexec/openssh/gnome-ssh-askpass
