From: Ian Campbell <ijc@hellion.org.uk>
Date: Thu, 17 Jan 2013 08:55:21 +0000
Subject: Tweak gitignore for Debian pkg-kernel using git
Forwarded: not-needed

[bwh: Tweak further for pure git]

Index: linux/.gitignore
===================================================================
--- linux.orig/.gitignore
+++ linux/.gitignore
@@ -76,11 +76,6 @@ modules.order
 #
 /rpmbuild/
 
-#
-# Debian directory (make deb-pkg)
-#
-/debian/
-
 #
 # Snap directory (make snap-pkg)
 #
@@ -91,19 +86,6 @@ modules.order
 #
 /tar-install/
 
-#
-# We don't want to ignore the following even if they are dot-files
-#
-!.clang-format
-!.cocciconfig
-!.editorconfig
-!.get_maintainer.ignore
-!.gitattributes
-!.gitignore
-!.kunitconfig
-!.mailmap
-!.rustfmt.toml
-
 #
 # Generated include files
 #
@@ -169,3 +152,10 @@ sphinx_*/
 
 # Rust analyzer configuration
 /rust-project.json
+
+#
+# Debian packaging: ignore everything at the top level, since it isn't
+# included in our repository
+#
+/*
+!/debian/
