---
 malloc/Makefile |    5 -----
 1 file changed, 5 deletions(-)

--- a/malloc/Makefile
+++ b/malloc/Makefile
@@ -261,10 +261,6 @@
 address-width=18
 endif
 
-# Unless we get a test for the availability of libgd which also works
-# for cross-compiling we disable the memusagestat generation in this
-# situation.
-ifneq ($(cross-compiling),yes)
 # If the gd library is available we build the `memusagestat' program.
 ifneq ($(LIBGD),no)
 others: $(objpfx)memusage
@@ -280,7 +276,6 @@
 # is to presume that the standard system headers will be ok for this file.
 $(objpfx)memusagestat.o: sysincludes = # nothing
 endif
-endif
 
 # Another goal which can be used to override the configure decision.
 .PHONY: do-memusagestat
